src/main/java/seedu/address/commons/core/CommandHistory.java

```diff
@@ -0,0 +1,77 @@
+package seedu.address.commons.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.util.AudioUtil;
+
+enum CommandHistoryType {
+    UNDO,
+    REDO
+}
+
+/**
+ * An abstraction for command history operations like viewing the previous and next command in history.
+ * Offers a similar experience to navigating through history of c ommands in a shell terminal..
+ */
+public class CommandHistory {
+    private final List<String> commandHistory;
+    private int currentCommandIndex = -1;
```

Comment:
should we use a magic word for -1?

Reply:
Perhaps we could do so in the future. But for now, I think it would just unnecessarily complicate things